### DISCUSSION TITLE
Update value of variables

### DISCUSSION THREAD

**Discussion by omidfathi** [FreeOpcUa/opcua-asyncio#1106] [Q&A]
Hi, how can I update the value of variables on the server side?

**Comment by AndreasHeine** [comment]
please look into the examples folder!

**Reply by omidfathi** [reply]
yes, I looked before. 
I use:
```python
await server.write_attribute_value(tags[987].nodeid, ua.DataValue(1.25))
```
and
print(tags[987].nodeid) --> NodeId(Identifier=520, NamespaceIndex=2, NodeIdType=<NodeIdType.FourByte: 1>)

but I get this warning :

Write refused: Variant: Variant(Value=1.25, VariantType=<VariantType.Double: 11>, Dimensions=None, is_array=False) with type VariantType.Double does not have expected type: VariantType.Float

**Reply by schroeder-** [reply]
The datatype must match. Try this:
`await server.write_attribute_value(tags[987].nodeid, ua.DataValue(ua.Float(1.25)))`

**Reply by omidfathi** [reply]
> The datatype must match. Try this: `await server.write_attribute_value(tags[987].nodeid, ua.DataValue(ua.Float(1.25)))`

Thank you, it's working now.
